src/viam/trajex/totg/trajectory.cpp

```diff
@@ -234,6 +243,14 @@ enum class integration_event : std::uint8_t {
     const double s_dot_new = s_dot + (s_ddot * dt);
     const arc_length s_new = s + arc_length{(s_dot * dt) + (0.5 * s_ddot * dt * dt)};
 
+    // Check that the step makes sufficient progress in the direction indicated by dt's sign.
+    // For positive dt (forward integration), s_new - s should be >= epsilon.
+    // For negative dt (backward integration), s_new - s should be >= -epsilon (i.e., |s_new - s| >= epsilon).
+    const double dt_sign = std::copysign(1.0, dt);
+    if ((s_new - s) < arc_length{epsilon * dt_sign}) [[unlikely]] {
```

Comment:
I think this isn't quite right for backwards yet. Consider, we are at s = 5 and we are moving backwards. We compute s_new = 3. Then (s_new - s) = -2. But the RHS will be -epsilon, and -2 is definitely less than that. So the exception will fire.